eng/doc/CrossPlatformCryptography.md

```diff
@@ -0,0 +1,336 @@
+# Cross-Platform Cryptography in Microsoft Go
+
+Cryptographic operations in Microsoft Go are delegated to the operating system (OS) libraries. This dependency has advantages:
+
+* Go apps benefit from OS reliability. Keeping cryptography libraries safe from vulnerabilities is a high priority for OS vendors. To do that, they provide updates that system administrators should be applying.
+* Go apps have access to FIPS-validated algorithms if the OS libraries are FIPS-validated.
+
+> [!NOTE]
+> Starting with Go 1.24, Go will also be FIPS 140-3 compliant, see https://github.com/golang/go/issues/69536.
+> If the only reason you are using Microsoft Go is for FIPS 140-3 compliance, you should consider using Microsoft Go 1.24 or later.
```

Comment:
This note seems a bit odd... if you are looking for FIPS 140-3 compliance, my intuition would be that you *must* use 1.24 or later?

My brain's autocorrect points me in another direction being the actual intent--to consider using upstream:

> If the only reason you are using Microsoft Go is for FIPS 140-3 compliance (not e.g. to conform to Microsoft policy), consider using upstream Go 1.24 or later.

Maybe there's just some missing context to connect how Microsoft Go, upstream Go, and "spec"-like Go 1.24 all relate to FIPS 140-3.

Reply:
Good catch, the last sentence should mention upstream Go, no Microsoft Go 😅. Will fix + add some clarification.